libfreerdp/core/transport.c

```diff
@@ -839,39 +884,61 @@ static int transport_default_read_pdu(rdpTransport* transport, wStream* s)
 	WINPR_ASSERT(transport);
 	WINPR_ASSERT(s);
 
-	/* Read in pdu length */
-	status = transport_parse_pdu(transport, s, &incomplete);
-	while ((status == 0) && incomplete)
+	/* RDS AAD Auth PDUs have no length indicator. We need to determine the end of the PDU by
+	 * reading in one byte at a time until we encounter the terminating null byte */
+	if (transport->AadMode)
 	{
+		BYTE c;
 		int rc;
-		if (!Stream_EnsureRemainingCapacity(s, 1))
-			return -1;
-		rc = transport_read_layer_bytes(transport, s, 1);
-		if (rc != 1)
-			return rc;
-		status = transport_parse_pdu(transport, s, &incomplete);
+		while (1)
```

Comment:
blocking loops are very problematic here as the user might cancel the connection, so check the `abortEvent` for being signalled

Reply:
It isn't blocking per se, it'll only read what's available. The only time it would be an issue is if the server sends a really really (really) long packet that takes time to consume